pkg/probing/fieldsequal.go

```diff
@@ -0,0 +1,43 @@
+package probing
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// FieldsEqualProbe checks if the values of the fields under the given json paths are equal.
```

Comment:
Do we want to enable docs linters for the probing package? I get that we do not want it in the main module but it could be nice here